src/hooks/useGasLimit.ts

```diff
@@ -60,12 +63,23 @@ const useGasLimit = (
   const [gasLimit, gasLimitError, gasLimitLoading] = useAsync<BigNumber>(() => {
     if (!safeAddress || !walletAddress || !encodedSafeTx || !web3ReadOnly) return
 
-    return web3ReadOnly.estimateGas({
-      to: safeAddress,
-      from: walletAddress,
-      data: encodedSafeTx,
-      type: operationType,
-    })
+    return web3ReadOnly
+      .estimateGas({
+        to: safeAddress,
+        from: walletAddress,
+        data: encodedSafeTx,
+        type: operationType,
+      })
+      .then((gasLimit) => {
+        // Due to a bug in Nethermind estimation, we need to increment the gasLimit by 30%
+        // when the safeTxGas is defined and not 0. Currently Nethermind is used only for Gnosis Chain.
+        const safeTxGas = safeTx?.data?.safeTxGas
+        if (currentChainId === chains.gno && safeTxGas && safeTxGas !== 0) {
+          return gasLimit.mul(130).div(100)
+        }
+
+        return gasLimit
+      })
   }, [safeAddress, walletAddress, encodedSafeTx, web3ReadOnly, operationType])
```

Comment:
Btw, strange that we're not getting lint errors here, since you're adding `safeTx` and `currentChainId` to the hook.